### PR TITLE
Allow uppercased plugins

### PR DESCRIPTION
**Pull request by benoitvidis**
Allows plugins to use uppercase letters in their names.

Actually, all checks related to elasticsearch index creation were already in place (handled in the plugin context). The PR only implements dedupe checks on plugin init.

### CODE REVIEW

File: lib/api/kuzzle.js [hunk 1]
```diff
@@ -44,11 +44,17 @@ const
...
   TokenManager = require('./core/auth/tokenManager'),
   Validation = require('./core/validation'),
-  InternalEngineBootstrap = require('../services/internalEngine/bootstrap'),
+  InternalBootstrap = require('../services/bootstrap/internalBootstrap'),
   Vault = require('./core/vault'),
   runShutdown = require('../util/shutdown'),
   Logger = require('../util/log');
 
+// Add meaningful stacktrace in Node.js 10
+// @todo must be removed in Node.js 12
```
Comment (kuzzle) [inline lib/api/kuzzle.js]: INFO Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

File: lib/util/esWrapper.js [hunk 2]
```diff
@@ -25,74 +25,89 @@ const
...
+    subcode: 'cannot_parse_mapping',
+    getPlaceholders: matches => [matches[1], matches[0]]
   },
   {
     // [index_not_found_exception] no such index, with { resource.type=index_or_alias & resource.id=foso & index=foso }
-    regex: /^\[index_not_found_exception] no such index, with { resource\.type=([^\s]+) (& )?resource\.id=([^\s]+) (& )?(index_uuid=.* )?index=([^\s]+) }$/,
-    replacement: 'Index "$3" does not exist, please create it first'
+    regex: /^no such index \[([%#])(.*)\/(.*)\]$/,
+    subcode: 'index_or_collection_not_found',
+    getPlaceholders: matches => {
```
Comment (kuzzle) [inline lib/util/esWrapper.js]: MINOR Remove curly braces and "return" from this arrow function body. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS3524)

File: lib/services/internalEngine/index.js [hunk 3]
```diff
@@ -300,153 +263,145 @@ class InternalEngine {
...
-    return this.exists(type, id)
-      .then(exists => {
+  update (collection, id, updateContent, options = {}) {
+    return this.exists(collection, id)
+      .then(({ body: exists }) => {
+        if (! exists) {
+          errorsManager.throw('document_not_found', id);
+        }
+
         const request = {
```
Comment (kuzzle) [inline lib/services/internalEngine/index.js]: MINOR Group all shorthand properties at either the beginning or end of this object declaration. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS3499)

**Comment by kuzzle** [comment]
SonarQube analysis reported 5 issues
* ![MINOR][MINOR] 3 minor
* ![INFO][INFO] 2 info

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [collectionEmulator.js#](https://github.com/kuzzleio/kuzzle/blob/95c495114f89858d353c536d85b4253e53d091d9/lib/util/collectionEmulator.js#): Add a new line at the end of this file. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AMissingNewlineAtEndOfFile)
1. ![INFO][INFO] [pluginsManager.js#L968](https://github.com/kuzzleio/kuzzle/blob/95c495114f89858d353c536d85b4253e53d091d9/lib/api/core/plugins/pluginsManager.js#L968): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by benoitvidis** [comment]
wrong target & duplicate

**Pull request closed without merging**